content/articles/generating-data-for-a-random-walk-using-python/index.md

```diff
@@ -0,0 +1,145 @@
+
+```python
+import matplotlib.pyplot as pylt
+from r_walk import RWalk
+#create a random walk
+ rwalk = RWalk()
+rwalk.fil_walk()
+
+p_numbers = list(range(rwalk.number_of_points)) #point numbers
+ pylt.scatter(rwalk.xvalues, rwalk.yvalues, c=p_numbers, cmap=plt.cm.Blues,
+ edgecolor='none', s=14)
+pylt.show()
+```
+First, we build a list of values corresponding to the amount of points in the path using `range()` function. Then we put them in a list called `p_numbers`, which we'll use to color every spot on the trek. We use the `Blues` colormap, supply `p_numbers` to the `c` parameter, and then `edgecolor=none` to remove the black accent around every spot.
+The result of the above program is shown below:
+![Random Walk with Blue colormaps](/engineering-education/generating-data-for-a-random-walk-using-python/Blue_colormap.PNG)
+
+#### Plotting the Starting and Ending Points
+Once the main sequence has been mapped, we can map the initial and last spots separately. To ensure the terminal points stand out, we'll enlarge them broader and color them distinctively, as illustrated below.
+Put this code just above the `pylt.show()` in the file `visual_rw.py`
+```python
+# map the initial and terminal points.
+ pylt.scatter(0, 0, c='green', edgecolors='none', s=110)
+ pylt.scatter(rwalk.xvalues[-1], rwalk.yvalues[-1], c='red', edgecolors='none',
+ s=110)
+```
+
+We paint spot `(0, 0)` in `green` with a greater size `(s=110)` than the other points to identify the beginning. The last `x and yvalues` in the trek are plotted in `red` with a size of `110` to indicate the end destination.
+When the above code is executed, you should be able to identify where the walk starts and terminates.
+
+### Adding Plot Points
+To offer us additional stuff to deal with, let's raise the number of dots. To achieve so, when creating an `RWalk` occurrence, we raise the number of `number_of_points` and alter the diameter of every dot when displaying the plot, as illustrated here:
+
+```python
+import matplotlib.pyplot as pylt
+from r_walk import RWalk
+#create a random walk
+ rwalk = RWalk(50000)
+rwalk.fil_walk()
+
+p_numbers = list(range(rwalk.number_of_points)) #point numbers
+ pylt.scatter(rwalk.xvalues, rwalk.yvalues, c=p_numbers, cmap=plt.cm.Blues,
+ edgecolor='none', s=1)
+pylt.show()
+```
+This sample generates a random walk with `50,000` spots and depicts every spot with a width of `s=1`. As demonstrated in the figure below, the consequent walk is feathery and cloud-like.
+
+![50000 points walk](/engineering-education/generating-data-for-a-random-walk-using-python/feathery.PNG)
+
+### Conclusion
+In this tutorial we have learned how to:
+ - Creating the RWalk() Class
+ - Choosing Directions
+ - Plotting the Random Walk
+ - Coloring the Points
+ - Plotting the Starting and Ending Points
+ - Adding Plot Points
+
+
+Happy coding!
```

Comment:
Add a `Further reading` section that contains additional references that the reader can learn from.